### PR TITLE
initial version

### PR DESCRIPTION
**Pull request by karakayasemi**

### CODE REVIEW

File: appinfo/info.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0"?>
+<info>
+    <id>analysis_app</id>
+    <name>Analysis App</name>
+    <description>Shows what your files types are</description>
+    <licence>AGPL</licence>
+    <author>Emre Küçük</author>
+    <dependencies>
+        <owncloud min-version="6.0.4" max-version="11.0.0"/>
```
Comment (karakayasemi) [inline appinfo/info.xml]: sen min version 9.0 max version 10.0

File: appinfo/info.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    <id>analysis_app</id>
+    <name>Analysis App</name>
+    <description>Shows what your files types are</description>
+    <licence>AGPL</licence>
+    <author>Emre Küçük</author>
+    <dependencies>
+        <owncloud min-version="6.0.4" max-version="11.0.0"/>
+    </dependencies>
+    <types>
+        <filesystem/>
```
Comment (karakayasemi) [inline appinfo/info.xml]: it is probably productivity check dev manual, find appropriate type

File: appinfo/routes.php [hunk 3]
```diff
@@ -0,0 +1,14 @@
+<?php
+
+return [
+    'routes' => [
+        // ui controller
+        ['name' => 'page#index', 'url' => '/', 'verb' => 'GET'],
+    //    ['name' => 'localApps#index', 'url' => '/installed-apps/{state}', 'verb' => 'GET', 'defaults' => ['state' => 'enabled']],
```
Comment (karakayasemi) [inline appinfo/routes.php]: remove comments

File: lib/AppInfo/Application.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        $container->registerService(
+            'URLGenerator', function(IContainer $c) {
+            /** @var \OC\Server $server */
+            $server = $c->query('ServerContainer');
+
+            return $server->getURLGenerator();
+        }
+        );
+
+        $container->registerService(
```
Comment (karakayasemi) [inline lib/AppInfo/Application.php]: indent!

File: lib/Controller/PageController.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
+<?php
```
Comment (karakayasemi) [inline lib/Controller/PageController.php:1]: this class should replace with controller/pagecontroller

File: appinfo/routes.php [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+return [
+    'routes' => [
+        ['name' => 'page#index', 'url' => '/', 'verb' => 'GET'],
+    ],
+    'resources' => []
```
Comment (karakayasemi) [inline appinfo/routes.php]: ?

File: lib/AppInfo/Application.php [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                $server = $c->query('ServerContainer');
+
+                 return $server->getURLGenerator();
+            }
+        );
+
+        $container->registerService(
+            'UserId', function(IContainer $c) {
+              $user = \OC::$server->getUserSession()
+                   ->getUser();
```
Comment (karakayasemi) [inline lib/AppInfo/Application.php]: indent

**Comment by karakayasemi** [comment]
please squash your commits

File: lib/Controller/PageController.php (hunk 5 above)
Comment (emrekuecuek) [reply lib/Controller/PageController.php:1]: contorller/pagecontroller.php has moved into lib/Controller/PageController.php

File: appinfo/routes.php (hunk 6 above)
Comment (emrekuecuek) [reply appinfo/routes.php]: 'resources' line has been deleted, also unnecessary commas has been deleted.

File: lib/AppInfo/Application.php (hunk 7 above)
Comment (emrekuecuek) [reply lib/AppInfo/Application.php]: has been done with spacebar, automatic indentation wouldn't do it properly.

Comment (karakayasemi) [reply lib/AppInfo/Application.php]: ?

File: lib/AppInfo/Application.php [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+                   return $user->getUID();
+               }
+            }
+        );
+
+        $container->registerService(
+            'L10N', function(IContainer $c) {
+                return $c->query('ServerContainer')
+                 ->getL10N($c->query('AppName'));
+                }
```
Comment (karakayasemi) [inline lib/AppInfo/Application.php]: indent

File: lib/AppInfo/Application.php [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+        $container->registerService(
+            'URLGenerator', function(IContainer $c) {
+            /** @var \OC\Server $server */
+                $server = $c->query('ServerContainer');
+
+                 return $server->getURLGenerator();
+            }
+        );
+
+        $container->registerService(
```
Comment (karakayasemi) [inline lib/AppInfo/Application.php]: inject user if it is necessary not userId

File: lib/AppInfo/Application.php [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+            'L10N', function(IContainer $c) {
+                return $c->query('ServerContainer')
+                 ->getL10N($c->query('AppName'));
+                }
+        );
+
+        $container->registerService(
```
Comment (karakayasemi) [inline lib/AppInfo/Application.php]: What is it for?

**Comment by karakayasemi** [comment]
If you register a service to DI container you should use it in somewhere. But, you are not using any of the services that you register.

File: lib/AppInfo/Application.php (hunk 9 above)
Comment (emrekuecuek) [reply lib/AppInfo/Application.php]: For now, app does not need any dependency so I cleaned inside of construct from them.

**Pull request merged**